spring-cloud-gcp-storage/src/main/java/org/springframework/cloud/gcp/storage/GoogleStorageResource.java

```diff
@@ -46,16 +47,32 @@
 
 	private final String location;
 
-	public GoogleStorageResource(Storage storage, String location) {
+	private boolean createBlobIfNotExists = true;
+
+	public GoogleStorageResource(Storage storage, String location,
+			boolean createBlobIfNotExists) {
 		Assert.notNull(storage, "Storage object can not be null");
 		this.storage = storage;
 		this.location = location;
+		this.createBlobIfNotExists = createBlobIfNotExists;
+	}
+
+	public GoogleStorageResource(Storage storage, String location) {
+		this(storage, location, true);
+	}
+
+	public boolean isCreateBlobIfNotExists() {
+		return this.createBlobIfNotExists;
+	}
+
+	public void setCreateBlobIfNotExists(boolean createBlobIfNotExists) {
+		this.createBlobIfNotExists = createBlobIfNotExists;
```

Comment:
I think we don't need this setter. The ctor arg is fully enough to avoid mutation in the future.
And, of course, needs to make `createBlobIfNotExists` property as `final`